odl/space/npy_tensors.py

```diff
@@ -1774,6 +1795,22 @@ def fallback_copy(x1, x2, n):
         x2_arr = x2.data
         out_arr = out.data
 
+    else:
+        # Need flat data for BLAS, otherwise in-place does not work.
+        # Raveling must happen in fixed order for non-contiguous out,
+        # otherwise 'A' is applied to arrays, which makes the outcome
+        # dependent on their respective contiguousness.
+        if out.data.flags.f_contiguous:
+            ravel_order = 'F'
+        else:
+            ravel_order = 'C'
```

Comment:
What if the array is not contiguous in any sense?

Reply:
Then we use C order. The F order thing is just for performance.
We need to flatten all arrays in the same fixed order, since otherwise the results will depend on the axis ordering of the inputs. I've had that bug :-)